### PR TITLE
Migrate to sbt-typelevel

### PR DESCRIPTION
**Pull request by armanbilge**
Fudged the versioning for now. Let's see how this fares in CI.

### CODE REVIEW

**Comment by armanbilge** [comment]
Ok, this is a good first step 👍

File: build.sbt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 val catsVersion = "2.6.1"
 
+ThisBuild / tlBaseVersion := "21.8" // TODO
```
Comment (bpholt) [inline build.sbt:3]: @armanbilge I noticed this TODO in this PR. Do you remember what was left to do here?

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,12 +1,7 @@
 addSbtPlugin("com.github.sbt" % "sbt-unidoc" % "0.5.0")
-addSbtPlugin("com.github.sbt" % "sbt-ci-release" % "1.5.9")
-addSbtPlugin("com.github.sbt" % "sbt-pgp" % "2.1.2")
+addSbtPlugin("org.typelevel" % "sbt-typelevel" % "0.4.3")
```
Comment (bpholt) [inline project/plugins.sbt:2]: This is now up to `0.4.9` if we want to update it before merging.

**Review by bpholt — COMMENTED** [review]
Other than the outstanding TODO, this looks good to me. @armanbilge are you still comfortable with it? If so, should I go ahead and merge, or would you like to make any more changes?

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt:3]: I'm afraid I don't remember the exact TODO I had in mind. I remember there was a larger discussion about what versioning scheme you wanted to use, so probably related to that.

File: project/plugins.sbt (hunk 2 above)
Comment (armanbilge) [reply project/plugins.sbt:2]: Yeah, let me update it.

File: build.sbt [hunk 3]
```diff
@@ -1,5 +1,8 @@
 val catsVersion = "2.7.0"
 
+ThisBuild / tlBaseVersion := "21.8" // TODO
+ThisBuild / tlCiReleaseBranches := Seq()
```
Comment (armanbilge) [inline build.sbt:4]: I also don't remember why I disabled snapshots.

File: build.sbt (hunk 1 above)
Comment (bpholt) [reply build.sbt:3]: Oh, that makes sense. I'm still not sure where we want to go with that, although I like the idea of semantic versioning with MiMa determining when there is a transitive breaking change. I think we can merge this without deciding that, for now.

Comment (armanbilge) [reply build.sbt:3]: Sounds good, I was able to get the MiMa thing going in https://github.com/typelevel/catbird/pull/430 :)

I remember why I disabled snapshots, b/c creds weren't setup yet. Let me re-enable them and then it's good to merge.

**Pull request merged**